### DISCUSSION TITLE
Option to define maximum time for download file

### DISCUSSION THREAD

**Discussion by primemb** [hgouveia/node-downloader-helper#112] [General]
Hi,
Is there any way to define the maximum amount of time one download can take for example if download take more than 10 minutes cancel and fire error event ?

**Comment by hgouveia** [comment]
@primemb hello, there is not way in the library, but nothing is stopping you to add that logic yourself, for example just to give you an idea,  you can just be checking the time passed during the progress event and if exceed your maximum time, just stop the download and handle it as you wish

**Comment by primemb** [comment]
Yes, i currently implemented that by setTimeout, but I think it is an excellent option if the package itself has an option for that
